WakaTime.py

```diff
@@ -279,14 +334,38 @@ def enough_time_passed(now, is_write):
             or (now - _last_hb['timestamp'] > (2 if is_write else HEARTBEAT_FREQUENCY * 60)))
 
 
+# removing path from the full filename... this should work under all OS
+def path_leaf(path):
+    head, tail = ntpath.split(path)
+    return tail or ntpath.basename(head)
+
+
 def handle_activity(is_write=False):
     global _last_hb
     if SHOW_KEY_DIALOG or not SETTINGS.get(settings, 'api_key', fallback=''):
         API_Key_Dialog.show()
     timestamp = time.time()
     last_file = _last_hb['entity'] if _last_hb is not None else ''
     if _filename and (_filename != last_file or enough_time_passed(timestamp, is_write)):
-        _last_hb = {'entity': _filename, 'timestamp': timestamp, 'is_write': is_write}
+        # use file- or folder-name to derive a project-name
+        blender_settings = bpy.context.preferences.addons[__name__].preferences
```

Comment:
What i'd do with this whole block up to the `_last_hb = {...` is to
* move it into a separate function
* cache it with [LRU decorator](https://docs.python.org/3/library/functools.html#functools.lru_cache), since it's called quite frequently with the same `_filename`, and is complex enough

Reply:
Did this.
However, just figured, that when the user changes AddOn-preferences, they will not change the project name until the user restarts Blender or loads/saves with a different filename. Doh!

Reply:
Hm... then we just have to pass the values of the relevant settings to the cached function.

Reply:
As you suggested, batched the changes and tested locally :+1: